File: day02/main.hs
import System.IO;

import Data.List;
import Data.List.Split;

processLine l = sort (map (read) (splitOn "x" l) :: [Integer])

processData d = map processLine (lines d) 

task1Core arr = 3 * a * b + 2 * a * c + 2 * b * c where [a,b,c] = arr

task2Core arr = a * b * c + 2 * a + 2 * b where [a,b,c] = arr

main = do
    contents <- readFile "data.txt"
    print $ sum (map task1Core (processData contents))
    print $ sum (map task2Core (processData contents))

File: day02/main.rb
R = File.read('data.txt').lines.map{|l| l.split('x').map{|n| n.to_i}.sort}

puts "Task 1: %d" % R.map{|a,b,c| 3*a*b + 2*a*c + 2*b*c }.inject(:+)
puts "Task 2: %d" % R.map{|a,b,c| 2*a+2*b + a*b*c}.inject(:+)